Sources/NIO/Codec.swift

```diff
@@ -346,6 +349,7 @@ public final class ByteToMessageHandler<Decoder: ByteToMessageDecoder> {
     }
 
     internal private(set) var decoder: Decoder? // only `nil` if we're already decoding (ie. we're re-entered)
+    private let maximumBufferSize: Int
```

Comment:
this should be stored as `Int?`

Reply:
no strong opinion, do you prefer to carry the optional and optional check?

Reply:
I think I'd do the optional: the optional check is not more expensive or syntactically heavy than the current -1 check you're doing now, and more accurately communicates the intent.